libbeat/kibana/client.go

```diff
@@ -206,14 +213,18 @@ func (conn *Connection) SendWithContext(ctx context.Context, method, extraPath s
 	req.Header.Set("Content-Type", "application/json")
 	req.Header.Add("Accept", "application/json")
 	req.Header.Set("kbn-xsrf", "1")
+	addHeaders(req.Header, conn.Headers)
+	addHeaders(req.Header, headers)
```

Comment:
nit: Should the `conn.Headers`  and `headers` be added before the `Content-Type` and `kbn-xsrf` are `set`?

Reply:
Good question. I'm not sure - I will defer to the Beats team. `headers` was already set after, so I left it there.

Reply:
I think I'd prefer to call addHeaders before setting the contents type. These are hard coded and should not be modifiable. well, in the end it doesn't really matter for most users.

Reply:
That sounds fine to me, thanks for weighing in. I'll update and add a test that checks that.

Reply:
Updated to set in order: headers configured in config, then in the function call, then statically defined ones (Content-Type, Accept, kbn-xsrf).

Note that I changed `req.Header.Add("Accept", ...` to `req.Header.Set("Accept", ...`, which means the header is always just one value.